modules/labs-react/search-form/lib/SearchForm.tsx

```diff
@@ -118,173 +117,318 @@ function getInputColors(theme: SearchThemeAttributes, isFocused?: boolean) {
-    '> div': {
-      display: 'block',
-    },
-  };
-});
-
-const SearchInput = styled(TextInput)<
-  Pick<SearchFormProps, 'isCollapsed' | 'grow' | 'height'> & {
-    inputColors: ReturnType<typeof getInputColors>;
-  }
->(({isCollapsed, inputColors, grow, height}) => {
-  const collapseStyles: CSSObject = isCollapsed
-    ? {
-        fontSize: '20px',
-        paddingLeft: `calc(${space.xl} + ${space.s})`,
-        paddingRight: `calc(${space.xl} + ${space.s})`,
-        maxWidth: 'none',
-        minWidth: 0,
-        backgroundColor: `rgba(0, 0, 0, 0)`,
-        height: height,
-      }
-    : {
-        maxWidth: grow ? '100%' : maxWidth,
-        minWidth: minWidth,
-        paddingLeft: `calc(${space.xl} + ${space.xxs})`,
-        paddingRight: space.xl,
-        backgroundColor: inputColors.background,
-        height: height,
-      };
-  return {
-    fontSize: '14px',
-    boxShadow: inputColors.boxShadow,
-    color: inputColors.color,
-    border: 'none',
-    WebkitAppearance: 'none',
-    transition: 'background-color 120ms, color 120ms, box-shadow 200ms, border-color 200ms',
-    zIndex: 2,
-    width: '100%',
-    '&::-webkit-search-cancel-button': {
       display: 'none',
     },
-    '&::placeholder': {
-      color: inputColors.placeholderColor,
+    [searchFieldPart]: {
+      width: '100%',
+      height,
+      maxWidth,
+      marginBottom: 0,
+      display: 'inline-block',
+      '> div': {
+        display: 'block',
+      },
     },
-    '&:placeholder-shown': {
-      textOverflow: 'ellipsis',
+    [`${submitSearchIconPart}, ${openSearchIconPart}`]: {
+      position: `absolute`,
+      margin: `auto ${system.space.x2}`,
+      top: 0,
+      bottom: 0,
+      left: 0,
+      padding: 0,
+      zIndex: 3,
+      ':dir(rtl)': {
+        right: 0,
+        left: 'auto',
+      },
     },
-    '&:not([disabled])': {
-      '&:focus, &:active': {
-        outline: 'none',
-        boxShadow: inputColors.boxShadow,
+    [searchInputPart]: {
+      maxWidth,
+      minWidth,
+      paddingInlineStart: calc.add(system.space.x10, system.space.x2),
+      paddingInlineEnd: system.space.x10,
+      backgroundColor: searchInputBackground,
+      height: height,
+      fontSize: system.fontSize.subtext.large,
+      boxShadow: searchInputBoxShadow,
+      color: searchInputColor,
+      border: 'none',
+      WebkitAppearance: 'none',
+      transition: 'background-color 120ms, color 120ms, box-shadow 200ms, border-color 200ms',
+      zIndex: 2,
+      width: '100%',
+      '&::-webkit-search-cancel-button': {
+        display: 'none',
+      },
+      '&::placeholder': {
+        color: searchInputPlaceholderColor,
       },
+      '&:placeholder-shown': {
+        textOverflow: 'ellipsis',
+      },
+
       '&:hover': {
-        backgroundColor: inputColors.backgroundHover,
+        backgroundColor: searchInputBackgroundHover,
+      },
+
+      '&:is(:focus-visible, &.focus):where(:not([disabled]))': {
```

Comment:
The `&` is not necessary inside an `&:is`.

```suggestion
      '&:is(:focus-visible, .focus):where(:not([disabled]))': {
```